### PR TITLE
Fix/drawer types source

### PR DESCRIPTION
**Pull request by Inbal-Tish**
## Description
Removing Drawer's manual typings and fixing generated types.
Adding 'customValue' prop to pass along from user to component and back on component's events.
Declaring props.

## Changelog
Drawer - Removing Drawer's manual typings and fixing generated types. Adding 'customValue' prop to pass along from user to component and back on component's events. Declaring missing props.

### CODE REVIEW

File: generatedTypes/index.d.ts [hunk 1]
```diff
@@ -35,6 +35,7 @@ export {default as TouchableOpacity, TouchableOpacityProps} from './components/t
 export {default as Button, ButtonPropTypes, ButtonProps} from './components/button';
 export {default as Checkbox, CheckboxPropTypes, CheckboxProps} from './components/checkbox';
 export {default as Chip, ChipPropTypes, ChipProps} from './components/chip';
+export {default as Drawer, DrawerProps, ItemProps} from './components/drawer';
```
Comment (ethanshar) [inline generatedTypes/index.d.ts:38]: We can't export a typing called `ItemProps` it's too vague. 
Better rename it to DrawerItemProps

File: generatedTypes/index.d.ts [hunk 2]
```diff
@@ -67,7 +68,6 @@ export {default as Carousel, CarouselProps} from './components/carousel';
 export {
   ActionSheet,
   ConnectionStatusBar,
-  Drawer,
```
Comment (ethanshar) [inline generatedTypes/index.d.ts:70]: Make sure to update the exports also in `src/index.ts` file

File: generatedTypes/index.d.ts (hunk 1 above)
Comment (Inbal-Tish) [reply generatedTypes/index.d.ts]: Done

File: generatedTypes/index.d.ts (hunk 2 above)
Comment (Inbal-Tish) [reply generatedTypes/index.d.ts:70]: Done

**Review by ethanshar — APPROVED** [review]

**Pull request merged**